### ISSUE TITLE
Kiali hangs on loading graphs, flooding logs with "port 0 is out of range" errors

### ISSUE THREAD

**Issue by taisph** [kiali/kiali#4379]
**Describe the bug**

Kiali has been running for about 24 hours and suddenly started hanging on loading graphs. The log is now flooding messages like below.
```
2021-09-21T09:27:23Z ERR Error stopping a port-forwarder: port 0 is out of range
2021-09-21T09:27:23Z ERR Error stopping a port-forwarder: port 0 is out of range
2021-09-21T09:27:23Z ERR Error stopping a port-forwarder: port 0 is out of range
2021-09-21T09:27:23Z ERR Error stopping a port-forwarder: port 0 is out of range
```

Kiali has issued a lot of warnings about not being able to proxy Istiod pods so that's probably related, but Kiali kept operating normally until now.
```
2021-09-21T09:27:23Z WRN GetPodProxyStatus is failing for [namespace: namespace] [pod: pod-name]: unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port 
```

**Versions used**

Kiali: 1.40.1
Istio: 1.9.8
Kubernetes flavour and version: v1.20.9-gke.701

**Comment by lucasponce** [comment]
Hi @taisph that's expected as Kiali 1.40.1 works with Istio 1.11 and higher versions.

Checkout the version compatibility:
https://kiali.io/documentation/latest/installation-guide/#_version_compatibility

And let us know if you could upgrade Istio or downgrade Kiali to a compatible version.

Thanks

**Comment by taisph** [comment]
I understand the versions are not supported but it's been working fine for a day now until this issue. And it works again after kicking the Kiali pod. And opening the firewall also solved the "GetPodProxyStatus" issue.

We will eventually upgrade istio. The dust needs to settle after "migrating" from Istio 1.5 last weekend.

**Comment by lucasponce** [comment]
Yes, I see, in short, we moved the proxy-status internal query to use an uniform port-forward in a long discussion and working PR
https://github.com/kiali/kiali/pull/4108

So, in short, 1.40.1 have non-compatible changes with Istio 1.9.

I think that Kiali 1.36.x may work in your environment, or at least it could be easier to adjust in your scenario, but we needed to bring this compatibility matrix as the Istio versions are growing and keeping the backward compatibility is a challenge.

I imagine, let us know if we can help you.
